### PR TITLE
docs: Reference rust doc style guide in contributing guidelines

### PR DESCRIPTION
**Pull request by flub**
## Description

It is good for us to follow the common rust documentation style guide,
so explicitly reference this guide.

## Notes & open questions

I'm effectively introducing a new policy here.  Feel free to discuss.

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- ~~[ ] Tests if relevant.~~
- ~~[ ] All breaking changes documented.~~

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Code contributions to Iroh are greatly appreciated. Here is the general workflow
 
    - **General code guidelines**
 
-     - When possible, please document relevant pieces of code. If documentation refers to another object (module, function, struct, etc) use ``[`path::to::ReferencedObject`]`` to link it. For more information check the [rustdoc docummentation][rustdoc]
+     - When possible, please document relevant pieces of code following the [rust documentation conventions][docconventtions]. For more information on how the rust documentation system works check the [rustdoc docummentation][rustdoc].
```
Comment (dignifiedquire) [inline CONTRIBUTING.md:37]: typo?

**Review by dignifiedquire — APPROVED** [review]
agree with using this as a reference guide 👍

File: CONTRIBUTING.md (hunk 1 above)
Comment (flub) [reply CONTRIBUTING.md]: i assume you meant the `[docconventtions]` with double `tt`?  Fixed that one, if there's another one let me know :)

Comment (dignifiedquire) [reply CONTRIBUTING.md]: yes, sorry I meant that one :)

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -13,6 +13,6 @@
 ## Change checklist
 
 - [ ] Self-review.
-- [ ] Documentation updates if relevant.
+- [ ] Documentation updates following the [style guide][https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#appendix-a-full-conventions-text], if relevant.
```
Comment (Frando) [inline .github/pull_request_template.md:16]:
```suggestion
- [ ] Documentation updates following the [style guide](https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#appendix-a-full-conventions-text), if relevant.
```
markdown link fix

Comment (Frando) [reply .github/pull_request_template.md:16]: Also maybe this is a nicer-to-read link to the same text:
https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text

Comment (flub) [reply .github/pull_request_template.md]: updated the link, thanks

**Pull request merged**